neqo-transport/src/packet.rs

```diff
@@ -156,6 +157,44 @@ impl PacketHdr {
     pub fn body_len(&self) -> usize {
         self.body_len
     }
+
+    // header length plus auth tag
+    pub fn overhead(&self, aead: &Aead, pmtu: usize) -> usize {
+        match &self.tipe {
+            PacketType::Short => {
+                let mut enc = Encoder::default();
+                // Leading byte.
+                let pnl = pn_length(self.pn);
+                enc.encode_byte(PACKET_BIT_SHORT | PACKET_BIT_FIXED_QUIC | encode_pnl(pnl));
+                enc.encode(&self.dcid.0);
+                enc.encode_uint(pnl, self.pn);
+                enc.len() + aead.expansion()
+            }
+            PacketType::VN(_) => encode_packet_vn(&self).len(),
```

Comment:
I don't think that we need this to return a valid value: VN might as well have infinite overhead, as does Retry.